R/scale-colour.R

```diff
@@ -78,13 +78,18 @@
 scale_colour_continuous <- function(...,
                                     type = getOption("ggplot2.continuous.colour")) {
   type <- type %||% "gradient"
+  args <- list2(...)
+  args$call <- args$call %||% current_call()
 
   if (is.function(type)) {
-    check_scale_type(type(...), "scale_colour_continuous", "colour")
+    if (!any(c("...", "call") %in% fn_fmls_names(type))) {
+      args$call <- NULL
+    }
+    check_scale_type(exec(type, !!!args), "scale_colour_continuous", "colour")
   } else if (identical(type, "gradient")) {
-    scale_colour_gradient(...)
+    exec(scale_colour_gradient, !!!args)
```

Comment:
while not major I think we should use inject over exec for reasons I can't quite formulate

Reply:
Yeah that was my intuition as well, but that ran into some problems I'm not skilled enough to debug properly.

If I change this to `inject(scale_colour_gradient(!!!args))` I get the following error: 
`Error: evaluation nested too deeply: infinite recursion / options(expressions=)?`.

So unless you have very strong feelings about this, I'd prefer to keep the `exec` variant.